### PR TITLE
dai: Remove unused DAI_TRIGGER_ definitions

### PR DESCRIPTION
**Pull request by paulstelian97**
I tried to use the definitions in dai.h to figure out how the DAI driver is being used from the framework, however these defines weren't compiling. This patch intends to fix this.

EDIT: From the discussion I ended up actually removing said definitions as they were unused and prone to the breakage I reported above. Using the COMP_TRIGGER_* definitions allows consistency in case of future updates.

Signed-off-by: Paul Olaru <paul.olaru@nxp.com>

### CODE REVIEW

File: src/include/sof/lib/dai.h [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <errno.h>
 #include <stddef.h>
 #include <stdint.h>
+#include <sof/audio/component.h> /* COMP_TRIGGER_* macros */
```
Comment (jajanusz) [inline src/include/sof/lib/dai.h]: We either comment why each of includes is needed or don't make comment like this at all. I'd remove that comment to be consistent.

Comment (paulstelian97) [reply src/include/sof/lib/dai.h]: Sure

Comment (jajanusz) [reply src/include/sof/lib/dai.h]: Thanks

**Comment by jajanusz** [comment]
@tlauda @paulstelian97 Why do we keep DAI_TRIGGER_* symbols at all ? - they are unused

**Comment by tlauda** [comment]
@jajanusz @paulstelian97 Clearly they are not needed, so let's remove them instead of trying to use them.

**Comment by paulstelian97** [comment]
I tried to use them for diagnosis. If you still believe they should be removed and the driver I'm currently writing (i.MX ESAI) should use the component defines directly, I'll go ahead and do the remove.

**Comment by jajanusz** [comment]
@paulstelian97 They are redundant anyway. It's better to user more abstract symbols (comp instead of dai).

**Comment by paulstelian97** [comment]
Should I modify this PR or create a new one for that?

**Comment by jajanusz** [comment]
@paulstelian97 You can modify this one.

**Pull request merged**